statsmodels/gam/tests/test_gam.py

```diff
@@ -0,0 +1,791 @@
+
+    univ_pol1 = UnivariatePolynomialSmoother(x[:, 0], degree=pol.degrees[0])
+    univ_pol2 = UnivariatePolynomialSmoother(x[:, 1], degree=pol.degrees[1])
+
+    gp1 = UnivariateGamPenalty(alpha=alphas[0], univariate_smoother=univ_pol1)
+    gp2 = UnivariateGamPenalty(alpha=alphas[1], univariate_smoother=univ_pol2)
+    mgp = MultivariateGamPenalty(multivariate_smoother=pol, alpha=alphas,
+                                 weights=weights)
+
+    for i in range(10):
+        params1 = np.random.randint(-3, 3, pol.smoothers[0].dim_basis)
+        params2 = np.random.randint(-3, 3, pol.smoothers[1].dim_basis)
+        params = np.concatenate([params1, params2])
+        c1 = gp1.func(params1)
+        c2 = gp2.func(params2)
+        c = mgp.func(params)
+        assert_allclose(c, c1 + c2, atol=1.e-10, rtol=1.e-10)
+
+        d1 = gp1.deriv(params1)
+        d2 = gp2.deriv(params2)
+        d12 = np.concatenate([d1, d2])
+        d = mgp.deriv(params)
+        assert_allclose(d, d12)
+
+        h1 = gp1.deriv2(params1)
+        h2 = gp2.deriv2(params2)
+        h12 = block_diag(h1, h2)
+        h = mgp.deriv2(params)
+        assert_allclose(h, h12)
+
+
+def test_generic_smoother():
+    x, y, poly = multivariate_sample_data()
+    alphas = [0.4, 0.7]
+    weights = [1, 1]
+
+    gs = GenericSmoothers(poly.x, poly.smoothers)
+    gam_gs = GLMGam(y, smoother=gs, alpha=alphas)
+    gam_gs_res = gam_gs.fit()
+
+    gam_poly = GLMGam(y, smoother=poly, alpha=alphas)
+    gam_poly_res = gam_poly.fit()
+
+    assert_allclose(gam_gs_res.params, gam_poly_res.params)
+
+
+def test_multivariate_gam_1d_data():
+    cur_dir = os.path.dirname(os.path.abspath(__file__))
+    file_path = os.path.join(cur_dir, "results", "prediction_from_mgcv.csv")
+    data_from_r = pd.read_csv(file_path)
+    # dataset used to train the R model
+    x = data_from_r.x.as_matrix()
+    y = data_from_r.y
+
+    df = [10]
+    degree = [3]
+    bsplines = BSplines(x, degree=degree, df=df)
+    # y_mgcv is obtained from R with the following code
+    # g = gam(y~s(x, k = 10, bs = "cr"), data = data, scale = 80)
+    y_mgcv = data_from_r.y_est
+
+    # alpha is by manually adjustment to reduce discrepancy in fittedvalues
+    alpha = [0.0168 * 0.0251 / 2 * 500]
+    gp = MultivariateGamPenalty(bsplines, alpha=alpha)
+
+    glm_gam = GLMGam(y, exog=np.ones((len(y), 1)), smoother=bsplines,
+                     alpha=alpha)
+#     res_glm_gam = glm_gam.fit(method='nm', max_start_irls=0,
+#                               disp=1, maxiter=10000, maxfun=5000)
+    res_glm_gam = glm_gam.fit(method='pirls', max_start_irls=0,
+                              disp=1, maxiter=10000)
+    y_gam = res_glm_gam.fittedvalues
+
+    # plt.plot(x, y_gam, '.', label='gam')
+    # plt.plot(x, y_mgcv, '.', label='mgcv')
+    # plt.plot(x, y, '.', label='y')
+    # plt.legend()
+    # plt.show()
+
+    assert_allclose(y_gam, y_mgcv, atol=0.01)
+
+
+def test_multivariate_gam_cv():
+    # SMOKE test
```

Comment:
pytest.mark.smoke?

Reply:
changed

Reply:
I dropped this again. pytest.mark.smoke is not yet available

Reply:
all it takes is one line in setup.cfg